store/storetest/user_store.go

```diff
@@ -3313,7 +3313,7 @@ func testUserStoreGetProfilesNotInTeam(t *testing.T, ss store.Store) {
 		Email:    MakeEmail(),
 		Username: "u2" + model.NewId(),
 	})).(*model.User)
-	defer func() { require.Nil(t, ss.User().PermanentDelete(u2.Id)) }()
```

Comment:
You are changing the behavior drastically here. The `defer` keyword defers the call of the function right after it until the end of the current function, in this case is used to cleanup the create user after all the tests are finished, if you remove the defer, it is called right away and remove the during your tests (and probably it will fail).